woocommerce/woocommerce
# Add support for SalePrice in JSON-LD schema for structured data

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #51997.

This update adds the `SalePrice`(https://schema.org/SalePrice) property in the JSON-LD schema for structured data. The regular price is moved under `ListPrice`(https://schema.org/ListPrice) property.

This enhancement aligns with recent changes to Google's structured data guidelines for merchant listings, providing more flexibility for representing sale prices. This allows merchants to more effectively showcase discounts and offers in search results.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Add a product.
2. Set a sale price.
3. Save and open the product permalink.
4. View page source code, search for `application/ld+json`.
5. Copy this script HTML element and check it with [Schema Markup Validator](https://validator.schema.org/)
6. Ensure both(sale and regular) prices are added

<img width="684" alt="image" src="https://github.com/user-attachments/assets/88727816-db48-4e42-9bb1-0e4345c6bf33">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

Add SalePrice property in JSON-LD schema for structured data.
</details>
